lib/resty/websocket/client.lua

```diff
@@ -210,16 +217,22 @@ function _M.connect(self, uri, opts)
         end
     end
 
-    local ok, err
-    if sock_opts then
-        ok, err = sock:connect(connect_addr, connect_port, sock_opts)
-    else
-        ok, err = sock:connect(connect_addr, connect_port)
-    end
+    local ok, err = sock:connect(connect_addr, connect_port, sock_opts)
     if not ok then
         return nil, "failed to connect: " .. err
     end
 
+    -- check for connections from pool:
+    local reused_count, err = sock:getreusedtimes()
+    if not reused_count then
+        return nil, "failed to get reused times: " .. tostring(err)
+    end
+
+    if reused_count > 0 then
+        -- being a reused connection (must have done handshake)
+        return 1
+    end
```

Comment:
Is this upstream fix even correct? It skips a duplicate TLS handshake on reuse (correct), but it _also_ skips the WebSocket handshake.